cmake/OpenCVDownload.cmake

```diff
@@ -37,6 +37,54 @@ file(WRITE "${OPENCV_DOWNLOAD_LOG}" "#use_cache \"${OPENCV_DOWNLOAD_PATH}\"\n")
 file(REMOVE "${OPENCV_DOWNLOAD_WITH_CURL}")
 file(REMOVE "${OPENCV_DOWNLOAD_WITH_WGET}")
 
+set(OPENCV_DOWNLOAD_MIRROR_ID "" CACHE STRING "Available mirror ID: gitcode, custom")
```

Comment:
Cache entry should be written below after "detection".

Here we should have:

```
ocv_check_environment_variables(OPENCV_DOWNLOAD_MIRROR_ID)
```

to support environment variables too.

Reply:
Done. It is indeed a good idea to use `ocv_check_environment_variables`, which I did not know before.